### PR TITLE
change: ETCM-8103 Save the referenced MC hash in sidechain pallet's storage

### PR DESCRIPTION
**Pull request by AmbientTea**
# Description

Adds a `LastMcHash` storage to `pallet-sidechain` and populates it via an inherent every time the referenced MC hash changes. This is so that:
* implementation of the native token observability can use it to query the range (old mc hash, new mc hash] for transactions
* there's better visibility into the system in general.

I've run it with 2 nodes and they produce a valid block.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages.
- [x] New tests are added if needed and existing tests are updated.
- [x] Relevant logging and metrics added
- [x] CI passes. See note on CI.
- [x] Any changes are noted in the `changelog.md` for affected crate
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: pallets/sidechain/src/lib.rs [hunk 1]
```diff
@@ -40,6 +56,11 @@ pub mod pallet {
 	pub(super) type SlotsPerEpoch<T: Config> =
 		StorageValue<_, sidechain_slots::SlotsPerEpoch, ValueQuery>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn last_mc_hash)]
+	pub(super) type LastMcHash<T: Config> =
+		StorageValue<_, sidechain_domain::McBlockHash, OptionQuery>;
```
Comment (mpskowron) [inline pallets/sidechain/src/lib.rs:62]: Would changing it to value query and adding a requirement to set it in genesis simplify things?

File: pallets/sidechain/src/tests.rs [hunk 2]
```diff
@@ -1,6 +1,12 @@
-use frame_support::traits::Hooks;
+use frame_support::{
+	pallet_prelude::{InherentData, ProvideInherent},
+	traits::Hooks,
+};
+use sidechain_domain::McBlockHash;
 
-use crate::mock::*;
+use crate::{mock::*, Call, LastMcHash, Pallet};
+use frame_support::traits::UnfilteredDispatchable;
```
Comment (LGLO) [inline pallets/sidechain/src/tests.rs:8]: Cosmetic: why are there two imports lists?

File: primitives/sidechain-mc-hash/Cargo.toml [hunk 3]
```diff
@@ -4,19 +4,39 @@ version = "0.1.0"
...
-sp-consensus-slots = { workspace = true }
-sidechain-domain = { workspace = true, features = ["std"] }
-sp-consensus = { workspace = true }
-sp-inherents = { workspace = true, features = ["std"] }
-sp-runtime = { workspace = true, features = ["std"] }
-sp-timestamp = { workspace = true }
-thiserror = { workspace = true }
+sp-inherents = { workspace = true }
+sidechain-domain = { workspace = true }
+
```
Comment (LGLO) [inline primitives/sidechain-mc-hash/Cargo.toml:13]: Tactical empty line?

File: primitives/sidechain-mc-hash/src/lib.rs [hunk 4]
```diff
@@ -1,229 +1,248 @@
...
+			inherent_data.put_data(INHERENT_IDENTIFIER, &self.mc_block.hash)
+		}
+
+		async fn try_handle_error(
+			&self,
+			identifier: &InherentIdentifier,
+			_error: &[u8],
+		) -> Option<Result<(), sp_inherents::Error>> {
+			if *identifier == INHERENT_IDENTIFIER {
+				panic!("BUG: {:?} inherent shouldn't return any errors", INHERENT_IDENTIFIER)
```
Comment (mpskowron) [inline primitives/sidechain-mc-hash/src/lib.rs:175]: It does return errors now, doesn't it?

**Review by mpskowron — COMMENTED** [review]
I'm not sold on why this change is needed. Runtime doesn't use this value at all, does it? Native token observability can just query it from header, can't it? Could you expand on "there's better visibility into the system in general."?

**Comment by AmbientTea** [comment]
Closing this for now since we can fetch this info from the header's digest.

**Pull request closed without merging**